### ISSUE TITLE
Does not work with Proxy Polyfill

### ISSUE THREAD

**Issue by jbackes** [gund/ng2-http-interceptor#41]
The module works perfectly on modern browsers, but for example for Safari 9, I need to use the proxy polyfill. Unfortunately, that does not work.
Safari says: EXCEPTION: Uncaught (in promise): TypeError: Proxy polyfill does not support trap 'http'
I think that's because of the private Http object in the service.

**Comment by gund** [comment]
Hi, thanks for reporting an issue, I will check it and try to find possible solutions.
Once I have a result - will update here.

**Comment by gund** [comment]
Hi @jbackes,

So for an update to take so long, I just managed time to get to this issue.
I investigated it and already came up with the fix will push in a while.

However I have some issues on Travis CI where all my builds fail (but should not) and all releases comes from CI server that is a blocker now (I have a bunch of pull requests also waiting to be merged).
So need to resolve that issue before this one can be merged and published.

Thank you for reporting a problem btw =)
